packages/cli/src/commands/hydrogen/dev.ts

```diff
@@ -156,3 +170,85 @@ async function runDev({
     },
   });
 }
+
+interface VariablesArguments {
+  root: string;
+  shop?: string;
+}
+async function combinedEnvironmentVariables({root, shop}: VariablesArguments) {
+  const remoteEnvironmentVariables = await pullRemoteEnvironmentVariables({
+    root,
+    flagShop: shop,
+    silent: true,
+  });
```

Comment:
This creates the following situation for those who didn't call `link`:

<img width="462" alt="image" src="https://github.com/Shopify/hydrogen/assets/1634092/6656819f-d9f1-4379-a69d-90debf75f0e8">

This seems to come from calling `ensureAuthenticatedAdmin` from the CLI kit.

We should not ask to log in when using the `dev` command, right? Just ignore the issue and continue without remote variables.